packages/flutter/test/widgets/focus_traversal_test.dart

```diff
@@ -849,5 +851,112 @@ void main() {
       expect(focusCenter.hasFocus, isFalse);
       expect(focusTop.hasFocus, isTrue);
     });
+    testWidgets('Focus traversal actions are invoked when shortcuts are used.', (WidgetTester tester) async {
+      final GlobalKey upperLeftKey = GlobalKey(debugLabel: 'upperLeftKey');
+      final GlobalKey upperRightKey = GlobalKey(debugLabel: 'upperRightKey');
+      final GlobalKey lowerLeftKey = GlobalKey(debugLabel: 'lowerLeftKey');
+      final GlobalKey lowerRightKey = GlobalKey(debugLabel: 'lowerRightKey');
+
+      await tester.pumpWidget(
+        WidgetsApp(
+          color: const Color(0xFFFFFFFF),
+          onGenerateRoute: (RouteSettings settings) {
+            return TestRoute(
+              child: Directionality(
+                textDirection: TextDirection.ltr,
+                child: FocusScope(
+                  debugLabel: 'scope',
+                  child: Column(
+                    children: <Widget>[
+                      Row(
+                        children: <Widget>[
+                          Focus(
+                            autofocus: true,
+                            debugLabel: 'upperLeft',
+                            child: Container(width: 100, height: 100, key: upperLeftKey),
+                          ),
+                          Focus(
+                            debugLabel: 'upperRight',
+                            child: Container(width: 100, height: 100, key: upperRightKey),
+                          ),
+                        ],
+                      ),
+                      Row(
+                        children: <Widget>[
+                          Focus(
+                            debugLabel: 'lowerLeft',
+                            child: Container(width: 100, height: 100, key: lowerLeftKey),
+                          ),
+                          Focus(
+                            debugLabel: 'lowerRight',
+                            child: Container(width: 100, height: 100, key: lowerRightKey),
+                          ),
+                        ],
+                      ),
+                    ],
+                  ),
+                ),
+              ),
+            );
+          },
+        ),
+      );
+
+      // Initial focus happens.
+      expect(Focus.of(upperLeftKey.currentContext).hasPrimaryFocus, isTrue);
+      await tester.sendKeyEvent(LogicalKeyboardKey.tab);
+      expect(Focus.of(upperRightKey.currentContext).hasPrimaryFocus, isTrue);
+      // Initial focus happens.
```

Comment:
Is this a copy/paste oversight? How is the initial focus happening again?

Reply:
Yeah, for some reason I CAN'T READ.  Removed all of the copies of that, and fixed up the first one (which was the only place that comment belonged).